File: 22-秋招/笔试/220827-360/1.py
n, m = list(map(lambda x:int(x), input().split(" ")))
nums = []
for i in range(n):
    nums.append(list(map(lambda x:int(x), input().split(" "))))

def solution(n, m, nums):
    cur = []
    all = 0
    sum = [0] * m
    for i in range(n):
        if i >= 7:
            cur.pop(0)
            for j in range(m):
                sum[j] = sum[j] - nums[i - 7][j]
        for j in range(m):
            sum[j] = sum[j] + nums[i][j]
        v = max(sum)
        s = sum(cur)
        if s < v:
            all += (v - s + 1)
            cur.append(v - s + 1)
        else:
            cur.append(0)

res = solution(n, m, nums)
print(res)

File: 22-秋招/笔试/其它/1.js
    temp = [0],
    rest = b - p[0];
  function getR(start, rest) {
    if (rest < p[start] || start >= n) {
      r.push(temp.slice(0));
      return;
    }

    for (let i = start; i < n; ++i) {
      temp.push(i);
      getR(i + 1, rest - p[i]);
      temp.pop();
    }
  }

  getR(1, rest);

  let res = Number.MAX_VALUE;

  function cal() {
    for (const x of r) {
      let sum = [];
      for (let i = 0; i < x.length - 1; ++i) {
        sum.push(t[i] * (x[i + 1] - x[i]));
      }
      sum.push(t[x.length - 1] * (n - x[x.length - 1]));
      res = Math.min(res, Math.max(...sum));
    }
  }
  cal();
  console.log(r);
  console.log(res);

  return res;
}

solution();
